src/phongo_bson_encode.c

```diff
@@ -545,3 +589,12 @@ void php_phongo_zval_to_bson_value(zval* data, php_phongo_bson_flags_t flags, bs
 	bson_destroy(&bson);
 	zval_ptr_dtor(&data_object);
 }
+
+void php_phongo_bson_append_zval(bson_t* bson, const char* key, long key_len, zval* value)
+{
+	php_phongo_field_path* field_path = php_phongo_field_path_alloc(false);
```

Comment:
This method was likely left over from a previous state where I also had an `append` method in the `BSON` class. This method has since been removed, but this function was left over. I removed it until I figure out whether it's needed (e.g. as part of a BSON writer).

As for your question, I took the logic from `php_phongo_zval_to_bson` and adapted it. I'd have to dive down into the logic myself to figure out why it does what.

Reply:
Noted that you've removed this function. FWIW, I do think this was correct as written. While you're appending fields, you can safely reference keys from the zval(s) you're working with. It makes sense that the field path would only need to do its own allocations for the decoding code path where it interprets a user-supplied type map.

Reply:
Feel free to resolve after reading.